octopoes/octopoes/models/ooi/reports.py

```diff
@@ -39,10 +39,25 @@ class Report(OOI):
 
     observed_at: datetime
     parent_report: Reference | None = ReferenceField("Report", default=None)
+    report_recipe: Reference | None = ReferenceField("ReportRecipe", default=None)
     has_parent: bool
 
     _natural_key_attrs = ["report_id"]
 
     @classmethod
     def format_reference_human_readable(cls, reference: Reference) -> str:
         return f"Report {reference.tokenized.report_id}"
+
+
+class ReportRecipe(OOI):
+    recipe_id: UUID
+    # object can be the total objects or 1 object, but then the ooi itself.
+    report_name_format: str
+    subreport_name_format: str
+    object_type: Literal["ReportRecipe"] = "ReportRecipe"
+    input_recipe: dict[str, Any]  # can contain a query which maintains a live set of OOIs or manually picked OOIs.
+    report_types: list[str]
+
+    @classmethod
+    def format_reference_human_readable(cls, reference: Reference) -> str:
+        return f"Report recipe {reference.tokenized.recipe_id}"
```

Comment:
I would say that the report_name_format should be part of the human readable. The id is not really readable.

Reply:
Ok then we need the instance name otherwise it would not make sense.